tests/protocol/test_wire.py

```diff
@@ -576,10 +586,46 @@ def http_get_handler(url, *_, **kwargs):
             self.assertEquals(success, False, "The download should have failed")
             self.assertEquals(len(urls), 4, "Unexpected number of HTTP requests: [{0}]".format(urls))
             self.assertEquals(urls[0], extension_url, "The first attempt should have been over the direct channel")
-            self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[1]), "The second attempt should have been over the host channel")
-            self.assertTrue(self.is_goal_state_request(urls[2]), "The host channel should have been refreshed the goal state")
-            self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[3]), "The third attempt should have been over the host channel")
-            self.assertEquals(HostPluginProtocol.is_default_channel(), False, "The host channel should not have been set as the default")
+            self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[1]),
+                            "The second attempt should have been over the host channel")
+            self.assertTrue(self.is_goal_state_request(urls[2]),
+                            "The host channel should have been refreshed the goal state")
+            self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[3]),
+                            "The third attempt should have been over the host channel")
+            self.assertEquals(HostPluginProtocol.is_default_channel(), False,
+                              "The host channel should not have been set as the default")
+
+    def test_download_ext_handler_pkg_should_fail_if_both_channels_return_an_unsuccessful_response(self):
```

Comment:
New test. This test would fail with the old code (tested it) since the download would "succeed" if the response was 404.